test/shared/createFederationNode.js

```diff
@@ -0,0 +1,182 @@
+import Fastify from 'fastify'
+import mercurius from 'mercurius'
+
+async function createNode(schema, resolvers) {
+  const app = Fastify()
+  app.register(mercurius, {
+    schema,
+    resolvers,
+    federationMetadata: true
+  })
+
+  app.post('/', async function () {
+    const query = '{ _service { sdl } }'
+    return app.graphql(query)
+  })
+
+  await app.listen({ port: 0 })
+  return [app, app.server.address().port]
+}
+
+export async function createNodeOne() {
+  const posts = {
+    p1: {
+      pid: 'p1',
+      title: 'Post 1',
+      content: 'Content 1',
+      authorId: 'u1'
+    },
+    p2: {
+      pid: 'p2',
+      title: 'Post 2',
+      content: 'Content 2',
+      authorId: 'u2'
+    },
+    p3: {
+      pid: 'p3',
+      title: 'Post 3',
+      content: 'Content 3',
+      authorId: 'u1'
+    },
+    p4: {
+      pid: 'p4',
+      title: 'Post 4',
+      content: 'Content 4',
+      authorId: 'u2'
+    }
+  }
+
+  const schema = `
+      #graphql
+      type Post @key(fields: "pid") {
+        pid: ID!
+        title: String
+        content: String
+        author: User @requires(fields: "pid title")
+      }
+  
+      type Query @extends {
+        topPosts(count: Int): [Post]
+      }
+  
+      type User @key(fields: "id") @extends {
+        id: ID! @external
+        name: String @external
+        posts: [Post]
+        numberOfPosts: Int @requires(fields: "id name")
+      }
+  
+      extend type Mutation {
+        createPost(post: PostInput!): Post
+        updateHello: String
+      }
+  
+      input PostInput {
+        title: String!
+        content: String!
+        authorId: String!
+      }
+    `
+
+  const resolvers = {
```

Comment:
The resolvers are not required for the tests. 
I think you can just pass a `{}` value.

In this way we can keep the code very clean.